test/yz_index_hashtree_eqc.erl

```diff
@@ -0,0 +1,553 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%-------------------------------------------------------------------
+
+%% @doc
+%% This module is a Quickcheck eqc_statem test that checks to make sure
+%% that the fundamental concepts of Yokozuna's AAE process implementation
+%% (i.e. start, insert, update, compare) operate correctly when inserting
+%% objects into both a single yokozuna hashtree process and a riak-kv hashtree
+%% process where n=1.
+
+%% Based on the differing objects between the trees, the riak-kv hashtree
+%% is the canonical source of the data and the count to repair is determined
+%% by objects that a yz_index_hashtree is missing.
+
+%% The goal of this test to start modelling Yokozuna's AAE implementation and
+%% to make sure it handles tree comparisons correctly.
+
+%% Key Things to Note:
+%% There's 1 riak_kv_idx_hashtree && 1 yz_kv_idx_hashtree per vnode.
+
+%% Each idx_hashtree can contain a collection of hashtrees...
+%% # of hashtrees per vnode == # in responsible_preflists
+
+%% This currently assumes N=1.
+
+%% We're sidestepping Yokozuna's logical partitioning
+%% (http://docs.basho.com/riak/latest/dev/advanced/search/#Indexes) for testing
+%% purposes.
+
+-module(yz_index_hashtree_eqc).
+-compile(export_all).
+
+-include("yokozuna.hrl").
+
+-ifdef(EQC).
```

Comment:
Noticed that some of the riak_kv eqc tests didn't, so I removed it. I'll bring 'em back!

Zeeshan Lakhani
programmer | 
software engineer at @basho | 
org. member/founder of @papers_we_love |
twitter => @zeeshanlakhani

> On Mar 13, 2015, at 6:11 PM, Sean Cribbs <notifications@github.com> wrote:
> 
> In test/yz_index_hashtree_eqc.erl:
> 
> > +
> > +%% Each idx_hashtree can contain a collection of hashtrees...
> > +%% # of hashtrees per vnode == # in responsible_preflists
> > +
> > +%% This currently assumes N=1.
> > +
> > +%% We're sidestepping Yokozuna's logical partitioning
> > +%% (http://docs.basho.com/riak/latest/dev/advanced/search/#Indexes) for testing
> > +%% purposes.
> > +
> > +-module(yz_index_hashtree_eqc).
> > +-compile(export_all).
> > +
> > +-include("yokozuna.hrl").
> > +
> > +-ifdef(EQC).
> While not strictly necessary, you should have an -ifdef(TEST). in this file.
> 
> —
> Reply to this email directly or view it on GitHub.
>